pkg/apis/stable/v1alpha1/gameserver.go

```diff
@@ -268,6 +259,45 @@ func (gs *GameServer) Validate() (bool, []metav1.StatusCause) {
 		}
 	}
 
+	return gs.validateConfig(causes)
+}
+
+// ValidateUpdate validates the GameServer configuration after the initial GameServer has been created.
+// If a GameServer is invalid there will be > 0 values in
+// the returned array
+func (gs *GameServer) ValidateUpdate() (bool, []metav1.StatusCause) {
+	var causes []metav1.StatusCause
+
+	// no host port when using dynamic PortPolicy only if we are still allocating a port.
+	if gs.Status.State == GameServerStatePortAllocation || gs.Status.State == "" {
```

Comment:
So I'm trying to work out how this passes, since, when we `PortAllocate`, we [actually set the `HostPort` values on the `Spec.Ports`](https://github.com/GoogleCloudPlatform/agones/blob/master/pkg/gameservers/portallocator.go#L147). (Which, maybe we shouldn't do? There is some precedent of editing `Spec` in `Pod` with `Spec.NodeName`)

But I think this leads to a larger question - Should we allow editing of a `GameServer.Spec` at all?
Editing the Spec should really drive restart/reschedule of the backing Pod, so we can actually accomplish what the declaration is asking for. Also, what if the `GameServer` is `Allocated` and then the Spec changes. This seems scary to me.

But I can't think of a use case in which we should be doing this. GameServers are deleted when they want to be updated, and new versions are rolled through a Fleet.

So I'm leaning towards not allowing Spec to be changed at all on Update.

Anyone else have other thoughts?

Reply:
make sense, this specific line needed a better comment also.

Reply:
Thanks for the feedback. The reason for the bug caught by the e2e tests is that Update validation runs anytime an update is done to the CRD either by kubectl or through state changes as the controller reconciles. I'm not sure if there's a way to just prevent manual updates. I couldn't find that.

Reply:
So I see how it passes :+1: I guess my question is - should we (a) remove that update to the Spec.Ports and then (b) lock down `Spec` so it can't be updated at all during the GameServer lifecycle.

I'm leaning this way, just because I can't think of any reason we would want to allow changes to GameServer.Spec once it's been created.